storm-core/src/jvm/org/apache/storm/stats/StatsUtil.java

```diff
@@ -1516,7 +1516,7 @@ public static ComponentPageInfo aggCompExecsStats(
 
             Integer start = ((Number) key.get(0)).intValue();
             String host = (String) value.get(0);
-            Integer port = (Integer) value.get(1);
+            Integer port = (Long) value.get(1);
```

Comment:
It's probably better to do this like in L1517. That way it doesn't matter if the actual object is an Integer or Long.